src/components/feedback-widget/feedback-widget.js

```diff
@@ -1,89 +1,81 @@
-import React, { useRef, useEffect, useState } from "react"
-import { css } from "@emotion/core"
-import styled from "@emotion/styled"
-import RatingOption from "./rating-option"
+import React, { useRef } from "react"
 import { post } from "axios"
+import { Machine, assign } from "xstate"
+import { useMachine } from "@xstate/react"
+import { OpenButton } from "./buttons"
+import FeedbackForm from "./feedback-form"
+import SubmitSuccess from "./submit-success"
+import SubmitError from "./submit-error"
+
+const postFeedback = ({ rating, comment }) => {
+  const payload = {
+    variables: {
+      rating,
+      comment,
+      url: window.location.href,
+    },
+    query: `
+      mutation submit($rating: Int!, $url: String!, $comment: String) {
+        submitFeedback(input: {
+          rating: $rating,
+          originUrl: $url,
+          comment: $comment
+        })
+      }
+    `,
+  }
 
-const buttonStyles = css`
-  background: rebeccapurple;
-  border: none;
-  border-radius: 0.5rem;
-  box-shadow: 1px 1px 6px #33333344;
-  color: white;
-  font-size: 14px;
-  font-family: sans-serif;
-  padding: 0.25rem 0.5rem;
-  z-index: 1;
-  -webkit-appearance: none;
-`
-
-const OpenButton = styled("button")`
-  display: block;
-  position: fixed;
-  bottom: 1rem;
-  right: 1rem;
-  ${buttonStyles};
-`
-
-const WidgetWrapper = styled("div")`
-  background-color: white;
-  border: 1px solid #ddd;
-  border-radius: 0.5rem;
-  bottom: 4rem;
-  font-family: sans-serif;
-  padding: 1rem;
-  position: fixed;
-  right: 1rem;
-  width: 300px;
-  z-index: 2;
-`
-
-const Fieldset = styled("fieldset")`
-  border: 0;
-  margin: 0 0 1rem;
-  padding: 0;
-`
-
-const TextareaLabel = styled("label")`
-  font-size: 0.875rem;
-  font-weight: bold;
-`
-
-const Textarea = styled("textarea")`
-  border: 1px solid #ddd;
-  display: block;
-  font-weight: normal;
-  margin: 0 0 1rem;
-  width: 100%;
-`
-
-const SubmitButton = styled("button")`
-  ${buttonStyles};
-  margin-right: 0.5rem;
-`
-
-const srOnly = css`
-  position: absolute;
-  width: 1px;
-  height: 1px;
-  padding: 0;
-  overflow: hidden;
-  -webkit-clip: rect(0, 0, 0, 0);
-  clip: rect(0, 0, 0, 0);
-  white-space: nowrap;
-  border: 0;
-`
+  return post(process.env.GATSBY_FEEDBACK_ENDPOINT, payload)
+}
 
-const CloseButton = styled("button")`
-  ${buttonStyles};
-  background: #ddd;
-  box-shadow: none;
-  color: inherit;
-  margin: 0.5rem 0;
-`
-const ScreenReaderText = styled("span")`
-  ${srOnly};
-`
+const feedbackMachine = Machine({
+  id: "feedback",
+  initial: "closed",
+  context: {
+    rating: 2,
+    comment: "",
+  },
+  states: {
+    opened: {
+      onEntry: "focusWidgetTitle",
+      on: {
+        RATE: {
+          actions: assign({ rating: (_ctx, e) => e.value }),
```

Comment:
What is `ctx`? I see it in the Xstate docs, but it's not super readable/intuitive for people new to it...

Reply:
Would this make it more clear?

```suggestion
          // Assign the RATE event value to the rating context
          actions: assign({ rating: (_context, event) => event.value }),
```

Reply:
Ooh yes, this is much better!

Reply:
Thanks for helping me realize the abbreviation can be confusing. Made the change docs-wide here: https://github.com/davidkpiano/xstate/commit/1d3215d2b5408e95ad79a51793d5d276275ce80f